### PR TITLE
Clean up client feature

### PR DESCRIPTION
**Pull request by dnewh**
This PR contains several updates to the client module. These changes are intended to work towards stabilizing the `client` feature. Changes include:
 - Standardizing documentation
 - Standardizing objects to derive the same traits where applicable
 - Removing the `create_client_factory` fn in favor of creating the factories directly
 - Reorganizing the client module to group like files and simplify config

### CODE REVIEW

File: sdk/src/client/purchase_order/reqwest.rs [hunk 1]
```diff
@@ -102,6 +102,7 @@ impl From<&PurchaseOrderRevisionDto> for PurchaseOrderRevision {
     }
 }
 
+/// The Reqwest implementation of the Location client
```
Comment (shannynalayna) [inline sdk/src/client/purchase_order/reqwest.rs:34]: s/location/Purchase Order

File: sdk/src/client/schema/reqwest.rs [hunk 2]
```diff
@@ -97,6 +97,7 @@ impl From<&DataTypeDto> for DataType {
     }
 }
 
+/// The Reqwest implementation of the Location client
```
Comment (shannynalayna) [inline sdk/src/client/schema/reqwest.rs]: s/location/Schema

File: sdk/src/client/purchase_order/reqwest.rs (hunk 1 above)
Comment (shannynalayna) [inline sdk/src/client/purchase_order/reqwest.rs:34]: s/location/Purchase Order

File: sdk/src/client/product/reqwest.rs [hunk 3]
```diff
@@ -87,6 +87,7 @@ impl From<&LatLongDto> for LatLong {
     }
 }
 
+/// The Reqwest implementation of the Location client
```
Comment (shannynalayna) [inline sdk/src/client/product/reqwest.rs]: s/ Location / Product

File: sdk/src/client/pike/reqwest.rs [hunk 4]
```diff
@@ -151,6 +151,7 @@ impl From<(&String, &String)> for InheritFrom {
     }
 }
 
+/// The Reqwest implementation of the Location client
```
Comment (shannynalayna) [inline sdk/src/client/pike/reqwest.rs]: s / Location / Pike

File: sdk/src/client/purchase_order.rs [hunk 5]
```diff
@@ -143,16 +140,20 @@ pub trait PurchaseOrderClient: Client {
     ///
     /// # Arguments
     ///
-    /// * `filter` - filter to apply to the list of `PurchaseOrder`s
-    fn list_purchase_orders(&self, filter: Option<&str>)
-        -> Result<Vec<PurchaseOrder>, ClientError>;
+    /// * `filters` - optional - filters to apply to the list of `PurchaseOrder`s
+    /// * `service_id` - optional service id if running splinter
```
Comment (shannynalayna) [inline sdk/src/client/purchase_order.rs:144]: Should / Could these updates to the comments go into the first commit ? Also, this `service_id` description doesn't match the ones from that commit

**Review by shannynalayna — DISMISSED** [review]
lgtm

File: sdk/src/client/reqwest/purchase_order/mod.rs [hunk 6]
```diff
@@ -239,14 +169,14 @@ impl PurchaseOrderClient for ReqwestPurchaseOrderClient {
         Ok(dto.iter().map(PurchaseOrderVersion::from).collect())
     }
```
Comment (leebradley) [inline sdk/src/client/reqwest/purchase_order/mod.rs:171]: Listing purchase order alternate ids is still unimplemented (line 268)

**Review by leebradley — CHANGES_REQUESTED** [review]

**Review by leebradley — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

**Pull request merged**